xbmc/windowing/gbm/OffScreenModeSetting.cpp

```diff
@@ -0,0 +1,50 @@
+#include "OffScreenModeSetting.h"
+
+bool COffScreenModeSetting::Init()
+{
+  // Open the render node that is needed for off-screen rendering
+  m_fd = open("/dev/dri/renderD128", O_RDWR);
+  return m_fd != -1;
+}
+
+void COffScreenModeSetting::Destroy()
+{
+  close(m_fd);
+  m_fd = -1;
+}
+
+std::vector<RESOLUTION_INFO> COffScreenModeSetting::GetModes()
+{
+    std::vector<RESOLUTION_INFO> resolutions;
+    RESOLUTION_INFO res;
```

Comment:
with `RESOLUTION_INFO res = GetCurrentMode();` you should be able to drop some duplicate code

Reply:
done